### PR TITLE
fix/6385

### LINKED ISSUE #6385: Sponsor registration: notes field can't be used on captive portal
**Describe the bug**
When doing sponsor registration on captive portal using portal modules, you can specify "notes" field as a mandatory field. It could be used to let user specify some motivations related to its registration.

However, when doing sponsor registration, PacketFence automatically write some informations into "Notes" field. Consequently, notes entered by user are lost.

**To Reproduce**
1. Create a portal module that use default sponsor source with "notes" as a mandatory field
2. Define a connection profile which use your new portal module
2. Add "Mark as sponsor" right to admin user
3. Try to register on captive portal using sponsor source with admin user 
4. Add some content into notes field
4. Check value of "notes" field in PacketFence web admin after node has been registered

=> It contains "sponsored .."

**Expected behavior**
We should keep notes entered by user.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Don't use notes for tracking the sponsered date.

# Impacts
The sponsor work flow.

# Issues
fixes #6385

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries

## Enhancements
* Track the sponspored date into the database.

### CODE REVIEW

File: html/pfappserver/root/src/views/Users/_config/index.js [hunk 1]
```diff
@@ -153,6 +154,13 @@ export const importFields = [
     required: false,
     validator: validatorFromColumnSchemas(MysqlDatabase.person.sponsor)
   },
+  {
+    value: 'sponsored_date',
+    text: i18n.t('Sponsor'),
```
Comment (JeGoi) [inline html/pfappserver/root/src/views/Users/_config/index.js]:
```suggestion
    text: i18n.t('Sponsored Date'),
```

File: html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/Sponsor.pm [hunk 2]
```diff
@@ -181,8 +181,8 @@ sub do_sponsor_registration {
     return unless($self->_validate_sponsor($sponsor));
 
     # form valid, adding person (using modify in case person already exists)
-    my $note = 'sponsored confirmation Date of arrival: ' . time2str("%Y-%m-%d %H:%M:%S", time);
-    $logger->info( "Adding guest person $pid" );
+    my $sponsered_date = time2str("%Y-%m-%d %H:%M:%S", time);
+    $logger->info( "Adding guest person $pid Date of arrival: $sponsered_date");
```
Comment (JeGoi) [inline html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/Sponsor.pm]:
```suggestion
    my $sponsored_date = time2str("%Y-%m-%d %H:%M:%S", time);
    $logger->info( "Adding guest person $pid Date of arrival: $sponsored_date");
```

File: html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/Sponsor.pm [hunk 3]
```diff
@@ -250,10 +250,10 @@ sub do_sponsor_registration {
 
     # update sponsor field with forced_sponsor value
     if (!defined($self->request_fields->{sponsor})) {
-        $self->update_person_from_fields(additionnal_fields => {notes => $note, sponsor => $sponsor});
+        $self->update_person_from_fields(additionnal_fields => {sponsered_date => $sponsered_date, sponsor => $sponsor});
```
Comment (JeGoi) [inline html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/Sponsor.pm]:
```suggestion
        $self->update_person_from_fields(additionnal_fields => {sponsored_date => $sponsored_date, sponsor => $sponsor});
```

File: html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/Sponsor.pm [hunk 4]
```diff
@@ -250,10 +250,10 @@ sub do_sponsor_registration {
...
     }
     else {
-        $self->update_person_from_fields(additionnal_fields => {notes => $note});
+        $self->update_person_from_fields(additionnal_fields => {sponsered_date => $sponsered_date});
```
Comment (JeGoi) [inline html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/Sponsor.pm]:
```suggestion
        $self->update_person_from_fields(additionnal_fields => {sponsored_date => $sponsored_date});
```

**Review by JeGoi — CHANGES_REQUESTED** [review]
Few minor issues with variables.
Then it is fine for me.
Thank you

File: db/upgrade-X.X-X.Y.sql [hunk 5]
```diff
@@ -49,6 +49,10 @@ call ValidateVersion;
 
 DROP PROCEDURE IF EXISTS ValidateVersion;
 
+\! echo "Updating the person table"
+ALTER TABLE person
+    ADD COLUMN sponsored_date DATETIME DEFAULT NULL;
```
Comment (fdurand) [inline db/upgrade-X.X-X.Y.sql]: IF NOT EXISTS

**Review by fdurand — CHANGES_REQUESTED** [review]

File: html/pfappserver/root/src/views/Users/_config/index.js (hunk 1 above)
Comment (jrouzierinverse) [reply html/pfappserver/root/src/views/Users/_config/index.js]: Done

**Review by JeGoi — APPROVED** [review]

**Review by JeGoi — APPROVED** [review]

**Review by fdurand — APPROVED** [review]

**Pull request merged**